### PR TITLE
Add support for custom elements using `:host(:hover)` like styling.

### PR DESCRIPTION
**Pull request by jpzwarte**
Screenshot 2021-06-04 at 10 16 40

### CODE REVIEW

**Comment by jpzwarte** [comment]
@ghengeveld Entirely possible that I broke other component types though; that's why I marked it "draft".

File: src/withPseudoState.js [hunk 1]
```diff
@@ -87,9 +87,12 @@ function rewriteStyleSheets(shadowRoot) {
                   states.push(`.pseudo-${state}`)
                   return ""
                 })
-                const stateSelector = shadowRoot
-                  ? `:host(${states.join("")}) ${plainSelector}`
-                  : `${states.join("")} ${plainSelector}`
+                let stateSelector;
+                if (shadowRoot && states.length) {
+                  stateSelector = `:host(${states.join('')}) ${selectorText.startsWith(':host') ? '' : plainSelector}`;
```
Comment (ghengeveld) [inline src/withPseudoState.js:92]: `selectorText` can contain several selectors (separated with a comma). Will `:host` only ever be on the first one? You should probably test against `selector` instead.

**Review by ghengeveld — APPROVED** [review]
Nice work. Code style isn't quite in line yet. Prettier should be able to auto-fix that.

File: src/withPseudoState.js (hunk 1 above)
Comment (jpzwarte) [reply src/withPseudoState.js]: Changed to `selector.includes(":host")`

**Comment by jpzwarte** [comment]
Ready to merge afaics.

**Comment by jpzwarte** [comment]
@ghengeveld I've also added support for https://developers.google.com/web/updates/2019/02/constructable-stylesheets
`LitElement` in Chrome uses this and without it, this add-on does not work.

**Comment by jpzwarte** [comment]
@ghengeveld I've reverted the constructible stylesheets commit. Since custom element stylesheets are shared between elements, the code enters a loop that freezes the browser. When `shadowRoot.adoptedStylesheets` are used, it should check if the *active* pseudo states have already been rewritten and do nothing in that case. But I think a larger refactor is needed for that. Moving that out of this PR.

**Comment by jpzwarte** [comment]
Fixed the loop. Stylesheets only get rewritten once.

**Comment by jpzwarte** [comment]
I improvement the state replacement code; this fixes #18

**Pull request merged**